libraries/freertos_plus/standard/freertos_plus_tcp/source/portable/NetworkInterface/Zynq/x_emacpsif_physpeed.c

```diff
@@ -163,19 +161,17 @@ int phy_detected = 0;
 
 static int detect_phy(XEmacPs *xemacpsp)
 {
-	u16 phy_reg;
-	u32 phy_addr;
-
-	for (phy_addr = 31; phy_addr > 0; phy_addr--) {
-		XEmacPs_PhyRead(xemacpsp, phy_addr, PHY_DETECT_REG,
-							&phy_reg);
-
-		if ((phy_reg != 0xFFFF) &&
-			((phy_reg & PHY_DETECT_MASK) == PHY_DETECT_MASK)) {
-			/* Found a valid PHY address */
-			FreeRTOS_printf( ("XEmacPs detect_phy: PHY detected at address %d.\r\n",
-																	phy_addr));
-			FreeRTOS_printf( ("XEmacPs detect_phy: PHY detected.\n" ) );
+	u16 id_lower, id_upper;
+	u32 phy_addr, id;
+
+	for (phy_addr = 0; phy_addr < 32; phy_addr++) {
```

Comment:
What's the significance of 32?

Reply:
There are [up to 32 phy slaves on the MDIO bus](https://en.wikipedia.org/wiki/Management_Data_Input/Output). To Dan's point, some links/pointers to applicable standards documents are helpful in code like this.  Also, if this is standard MDIO code we could probably make it a common module at some point, but that's beyond the scope of this PR.